mcs/tools/mdoc/Mono.Documentation/monodocer.cs

```diff
@@ -1654,18 +1670,25 @@ public void UpdateType (XmlElement root, TypeDefinition type)
 
 	/// <summary>Adds an AssemblyInfo with AssemblyName node to an XmlElement.</summary>
 	/// <returns>The assembly that was either added, or was already present</returns>
-	static XmlElement AddAssemblyNameToNode (XmlElement root, TypeDefinition type)
+	XmlElement AddAssemblyNameToNode (XmlElement root, TypeDefinition type)
 	{
 		return AddAssemblyNameToNode (root, type.Module);
 	}
 
 	/// <summary>Adds an AssemblyInfo with AssemblyName node to an XmlElement.</summary>
 	/// <returns>The assembly that was either added, or was already present</returns>
-	static XmlElement AddAssemblyNameToNode (XmlElement root, ModuleDefinition module)
+	XmlElement AddAssemblyNameToNode (XmlElement root, ModuleDefinition module)
 	{
 		Func<XmlElement, bool> assemblyFilter = x => {
 			var existingName = x.SelectSingleNode ("AssemblyName");
-			return existingName != null && existingName.InnerText == module.Assembly.Name.Name;
+
+			bool apiStyleMatches = true;
+			string currentApiStyle = x.GetAttribute ("apistyle");
+			if ((HasDroppedNamespace (module) && !string.IsNullOrWhiteSpace (currentApiStyle) && currentApiStyle != "unified") ||
+				(isClassicRun && (string.IsNullOrWhiteSpace (currentApiStyle) || currentApiStyle != "classic"))) {
```

Comment:
Style: This line should be indented an extra tab stop so that it doesn't look like `if` block contents